File: util/test_runner.py
import io
import os
import sys
from typing import Callable
import subprocess


def run_fun_solution(input_data: str, func: Callable[[], None]) -> str:
  backup_stdin = sys.stdin
  backup_stdout = sys.stdout
  try:
    sys.stdin = io.StringIO(input_data)
    sys.stdout = io.StringIO()
    func()
    return sys.stdout.getvalue()
  finally:
    sys.stdin = backup_stdin
    sys.stdout = backup_stdout


def run_solution(input_data: str, file_path:str, file_name: str) -> str:
  result = subprocess.run(
    [sys.executable, os.path.join(os.path.dirname(file_path), file_name)],
    input=input_data,
    text=True,
    capture_output=True
  )

  return result.stdout

File: util/test_runner.js
const {spawn} = require('child_process');
const path = require('path');

function runSolution(input, filePath, fileName) {
  return new Promise((resolve, reject) => {
    const solutionPath = path.join(filePath, fileName);
    const proc = spawn('node', [solutionPath]);

    let stdout = '';
    let stderr = '';

    proc.stdout.on('data', (data) => {
      stdout += data;
    });

    proc.stderr.on('data', (data) => {
      stderr += data;
    });

    proc.on('error', (err) => {
      reject(err);
    });

    proc.on('close', (code) => {
      if (code !== 0) {
        reject(new Error(stderr));
      } else {
        resolve(stdout);
      }
    });

    proc.stdin.write(input);
    proc.stdin.end();
  });
}

module.exports = {runSolution};